src/net_processing.cpp

```diff
@@ -3943,11 +3969,6 @@ bool PeerManagerImpl::ProcessMessages(CNode* pfrom, std::atomic<bool>& interrupt
         if (!peer->m_getdata_requests.empty()) return true;
     }
 
-    {
-        LOCK(g_cs_orphans);
-        if (!peer->m_orphan_work_set.empty()) return true;
```

Comment:
I think this introduce a lightweight behavior change.

Previously, if our peer has `fDisconnect=true` set up from a previous message processing and the orphan work set is still non-empty after the `ProcessOrphanTx` above, we would have return `false`.

From now on, if the orphan work set is still non-empty after `ProcessOrphanTx`, we're going to return `true`. 

Though i don't think it has any impact on allowing our peer to abuse more our CPU time, when we return from `ProcessMessages` to `ThreadMessageHandler`, we go to `SendMessages` which calls `MaybeDiscourageAndDisconnect` as first thing.

(Actually why the check on `fDisconnect` isn't first in `ProcessMessages`, should be before any kind of processing work attempted with `ProcessGetData`/`ProcessOrphanTx` ?)

Reply:
`fDisconnect` is already checked in net.cpp prior to `ProcessMessages` being called, but there's always a chance that there's a race and `fDisconnect` is set while in the middle of `ProcessMessages` for the peer.